File: PHP/Program/Clothing.php
<?php
require ('Product.php');

class Clothing extends Product{
    private $size;
    private $material;
    private $gender;
    
    function __construct($id, $nama, $brand, $harga, $size, $material, $gender) {
        parent::__construct($id, $nama, $brand, $harga);
        $this->size = $size;
        $this->material = $material;
        $this->gender = $gender;
    }

        // Getter methods
        function getSize() {
            return $this->size;
        }
    
        function getMaterial() {
            return $this->material;
        }
    
        function getGender() {
            return $this->gender;
        }
    
        // Setter methods
        function setSize($size) {
            $this->size = $size;
        }
    
        function setMaterial($material) {
            $this->material = $material;
        }
    
        function setGender($gender) {
            $this->gender = $gender;
        }
}


?>

File: CPP/Program/Clothing.cpp
#include <bits/stdc++.h>
#include "Product.cpp"

using namespace std;

class Clothing : public Product{
    private:
        string size;
        string material;
        string gender;
    public:

        Clothing() : Product(){
            this->size = "";
            this->material = "";
            this->gender = "";
        }

        Clothing(int id, string nama, string brand, int harga, string size, string material, string gender) : Product(id, nama, brand, harga){
            this->size = size;
            this->material = material;
            this->gender = gender;
        }

        // Getter
        string getSize() { 
            return size; 
        }
        string getMaterial() { 
            return material; 
        }
        string getGender() { 
            return gender; 
        }

        // Setter
        void setSize(string size) { 
            this->size = size; 
        }
        void setMaterial(string material) { 
            this->material = material; 
        }
        void setGender(string gender) { 
            this->gender = gender; 
        }

        ~Clothing(){
            
        }
};
